napari/components/_tests/test_world_coordinates.py

```diff
@@ -104,5 +104,4 @@ def test_warning_affine_slicing():
     assert 'Non-orthogonal slicing is being requested' in str(wrn[0].message)
     with pytest.warns(UserWarning) as recorded_warnings:
         viewer.layers[0].refresh()
-    # note right now refresh tiggers two `_slice_indices` calls
-    assert len(recorded_warnings) == 2
+    assert len(recorded_warnings) == 1
```

Comment:
The refactor here means we can skip one call to `_slice_indices` by storing the result in a local variable, giving us a minor performance boost in some cases.